daemon/container_operations.go

```diff
@@ -115,6 +115,16 @@ func (daemon *Daemon) buildSandboxOptions(container *container.Container) ([]lib
 			return nil, err
 		}
 		parts := strings.SplitN(extraHost, ":", 2)
+		// If the IP Address is a string called "host-gateway", replace this
+		// value with the IP address stored in the daemon level HostGatewayIP
+		// config variable
+		if parts[1] == network.HostGatewayName {
+			if gateway := daemon.configStore.HostGatewayIP; net.ParseIP(gateway) != nil {
+				parts[1] = gateway
+			} else {
+				logrus.Warnf("HostGatewayIP value %s is invalid", gateway)
```

Comment:
This should never happen if the config is validated on startup and HostGatewayIP is already a `net.IP`

Reply:
there can be a case where default bridge is disabled and the HostGatewayIP is not set, in that case might be best to bail out ?